ArchiveHunterDownloadManager/AppDelegate.m

```diff
@@ -429,4 +446,25 @@ - (NSApplicationTerminateReply)applicationShouldTerminate:(NSApplication *)sende
     return NSTerminateNow;
 }
 
+- (void) getEntities:(NSManagedObject *)bulk {
+    NSString *token = [self getReterievalToken:bulk];
+    NSString *serverType = [self getServerType:bulk];
+    NSString *serverHost = [self hostNameForServerSource:serverType];
+    NSURL *apiURL = [NSURL URLWithString:[NSString stringWithFormat:@"https://%@/api/bulkv2/%@/summarystream", serverHost, token]];
```

Comment:
From the function definitions above, `*token` and `*serverType` are both nullable - so  before using them here you should check if either is null and return early from `getEntities`, logging out a message with NSLog

Reply:
Done.